src/garage/experiment/snapshotter.py

```diff
@@ -1,127 +1,125 @@
         return self._snapshot_gap
 
-    @snapshot_gap.setter
-    def snapshot_gap(self, gap):
-        self._snapshot_gap = gap
-
     def save_itr_params(self, itr, params):
         """Save the parameters if at the right iteration."""
-        if self._snapshot_dir:
-            file_name = None
-
-            if self._snapshot_mode == 'all':
-                file_name = osp.join(self._snapshot_dir, 'itr_%d.pkl' % itr)
-            elif self._snapshot_mode == 'last':
-                # override previous params
-                file_name = osp.join(self._snapshot_dir, 'params.pkl')
-            elif self._snapshot_mode == 'gap':
-                if itr % self._snapshot_gap == 0:
-                    file_name = osp.join(self._snapshot_dir,
+        file_name = None
+
+        if self._snapshot_mode == 'all':
+            file_name = os.path.join(self._snapshot_dir, 'itr_%d.pkl' % itr)
+        elif self._snapshot_mode == 'last':
+            # override previous params
+            file_name = os.path.join(self._snapshot_dir, 'params.pkl')
+        elif self._snapshot_mode == 'gap':
+            if itr % self._snapshot_gap == 0:
+                file_name = os.path.join(self._snapshot_dir,
                                          'itr_%d.pkl' % itr)
-            elif self._snapshot_mode == 'gap_and_last':
-                if itr % self._snapshot_gap == 0:
-                    file_name = osp.join(self._snapshot_dir,
+        elif self._snapshot_mode == 'gap_and_last':
+            if itr % self._snapshot_gap == 0:
+                file_name = os.path.join(self._snapshot_dir,
                                          'itr_%d.pkl' % itr)
-                file_name_last = osp.join(self._snapshot_dir, 'params.pkl')
-                with open(file_name_last, 'wb') as file:
-                    pickle.dump(params, file)
-            elif self._snapshot_mode == 'none':
-                pass
-            else:
-                raise ValueError('Invalid snapshot mode {}'.format(
-                    self._snapshot_mode))
-
-            if file_name:
-                with open(file_name, 'wb') as file:
-                    pickle.dump(params, file)
-
-    def load(self, itr='last'):
-        r"""Load one snapshot of parameters from disk.
+            file_name_last = os.path.join(self._snapshot_dir, 'params.pkl')
+            with open(file_name_last, 'wb') as file:
+                pickle.dump(params, file)
+        elif self._snapshot_mode == 'none':
+            pass
+        else:
+            raise ValueError('Invalid snapshot mode {}'.format(
+                self._snapshot_mode))
+
+        if file_name:
+            with open(file_name, 'wb') as file:
+                pickle.dump(params, file)
+
+    def load(self, resume_from_dir, itr='last'):
+        """Load one snapshot of parameters from disk.
 
         Args:
-            itr(int or string): Iteration to load.
+            resume_from_dir (str): Directory of the pickle file
```

Comment:
this is API leakage. we don't know we are resuming anything. this component only loads things.